docs/user-guide/arrays.rst

```diff
@@ -0,0 +1,528 @@
+
+A Zarr array can be resized, which means that any of its dimensions can be
+increased or decreased in length. For example:
+
+.. ipython:: python
+
+   z = zarr.zeros(
+      store="data/example-4.zarr",
+      shape=(10000, 10000),
+      chunks=(1000, 1000)
+   )
+   z[:] = 42
+   z.shape
+   z.resize((20000, 10000))
+   z.shape
+
+Note that when an array is resized, the underlying data are not rearranged in
+any way. If one or more dimensions are shrunk, any chunks falling outside the
+new array shape will be deleted from the underlying store.
+
+:func:`zarr.Array.append` is provided as a convenience function, which can be
+used to append data to any axis. E.g.:
+
+.. ipython:: python
+
+   a = np.arange(10000000, dtype='i4').reshape(10000, 1000)
+   z = zarr.array(store="data/example-5", data=a, chunks=(1000, 100))
+   z.shape
+   z.append(a)
+   z.append(np.vstack([a, a]), axis=1)
+   z.shape
+
+.. _tutorial_compress:
+
+Compressors
+-----------
+
+A number of different compressors can be used with Zarr. A separate package
+called NumCodecs_ is available which provides a common interface to various
+compressor libraries including Blosc, Zstandard, LZ4, Zlib, BZ2 and
+LZMA. Different compressors can be provided via the ``compressor`` keyword
+argument accepted by all array creation functions. For example:
+
+.. ipython:: python
+
+   from numcodecs import Blosc
+
+   compressor = None  # TODO: Blosc(cname='zstd', clevel=3, shuffle=Blosc.BITSHUFFLE)
+   data = np.arange(100000000, dtype='i4').reshape(10000, 10000)
+   # TODO: remove zarr_format
+   z = zarr.array(store="data/example-6.zarr", data=data, chunks=(1000, 1000), compressor=compressor, zarr_format=2)
+   None  # TODO: z.compressor
+
+This array above will use Blosc as the primary compressor, using the Zstandard
+algorithm (compression level 3) internally within Blosc, and with the
+bit-shuffle filter applied.
+
+When using a compressor, it can be useful to get some diagnostics on the
+compression ratio. Zarr arrays provide the :func:`zarr.Array.info` property
+which can be used to print useful diagnostics, e.g.:
+
+.. ipython:: python
+
+   z.info
+
+The :func:`zarr.Array.info_complete` method inspects the underlying store and
+prints additional diagnostics, e.g.:
+
+.. ipython:: python
+
+   z.info_complete()
+
+If you don't specify a compressor, by default Zarr uses the Blosc
+compressor. Blosc is generally very fast and can be configured in a variety of
+ways to improve the compression ratio for different types of data. Blosc is in
+fact a "meta-compressor", which means that it can use a number of different
+compression algorithms internally to compress the data. Blosc also provides
+highly optimized implementations of byte- and bit-shuffle filters, which can
+improve compression ratios for some data. A list of the internal compression
+libraries available within Blosc can be obtained via:
+
+.. ipython:: python
+
+   from numcodecs import blosc
+
+   blosc.list_compressors()
+
```

Comment:
The default now is Zstd. Also, I think we should link out for the details of the blosc codec.